src/config/configJson.ts

```diff
@@ -12,17 +12,23 @@ export default class ConfigJsonConfigBackend {
 	private readonly writeLockConfigJson: () => Bluebird.Disposer<() => void>;
 
 	private readonly schema: Schema.Schema;
+	/**
+	 * @deprecated configPath is only set by legacy tests
+	 */
 	private readonly configPath?: string;
 
 	private cache: { [key: string]: unknown } = {};
 
-	private readonly init = _.once(async () =>
-		Object.assign(this.cache, await this.read()),
-	);
+	private readonly init = _.once(async () => {
+		if ((await osRelease.getOSVersion(constants.hostOSVersionPath)) == null) {
```

Comment:
Why move this check here? Getting the OS version was only there to know where to look for config.json no?

Reply:
Actually, I'm not sure of the point of the check

Reply:
I'm not sure either but was iffy about removing it. Looking at the section though, the check could be the determine that config.json exists when initializing `ConfigJsonConfigBackend`, because OS version should always be present in config.json on first boot. Since that's the case, we can remove this check and just allow `await this.read()` to fail if config.json doesn't exist.